specs/001-lrt-current-release/plan.md

```diff
@@ -0,0 +1,143 @@
+│   └── scripts/
+│       └── manage-radius-installation.sh  # New: version detection and install/upgrade logic
+└── Makefile                           # No changes needed (tests run via existing make targets)
+```
+
+**Structure Decision**: Minimal changes - single workflow file modification plus a helper script for version management logic. The helper script follows the principle of extracting complex logic from workflow YAML into testable shell scripts.
+
+## Complexity Tracking
+
+No constitution violations. This change reduces complexity by removing:
+
+- Build job (~200 lines)
+- Skip-build logic (~50 lines)
+- Caching logic (~40 lines)
+- Build-related environment variables (~10 lines)
+
+## Post-Design Constitution Re-Check
+
+*Re-evaluated after Phase 1 design completion.*
+
+| Principle                           | Status  | Post-Design Notes                                                              |
+|-------------------------------------|---------|--------------------------------------------------------------------------------|
+| I. API-First Design                 | ✅ Pass | No API changes                                                                 |
+| II. Idiomatic Code Standards        | ✅ Pass | Shell script follows `set -euo pipefail`, proper quoting, clear variable names |
+| III. Multi-Cloud Neutrality         | ✅ Pass | Test infrastructure only, no cloud-specific dependencies added                 |
+| IV. Testing Pyramid Discipline      | ✅ Pass | Functional tests validate the workflow; manual verification steps documented   |
+| V. Collaboration-Centric Design     | ✅ Pass | Simplified workflow benefits all contributors                                  |
+| VI. Open Source and Community-First | ✅ Pass | Design fully documented in design-notes repository                             |
+| VII. Simplicity Over Cleverness     | ✅ Pass | Net reduction of ~300 lines; uses standard CLI commands                        |
+| VIII. Separation of Concerns        | ✅ Pass | Version logic in shell script, workflow handles orchestration                  |
+| IX. Incremental Adoption            | ✅ Pass | No breaking changes to other workflows or user-facing behavior                 |
+
+**Gate Status**: ✅ PASSED - Ready for task breakdown (/speckit.tasks)
+
+## Implementation Phases
+
+### Phase 1: Setup (Shared Infrastructure)
+
+- Create `manage-radius-installation.sh` helper script
+- Implement version detection logic (parse rad version output)
+- Implement conditional install/upgrade logic
+- Add error handling for upgrade failures
+
+### Phase 2: Remove Build Job (User Story 4)
+
+- Delete the entire `build` job from workflow
+- Remove build-related environment variables
+- Update workflow file header comments
+
+### Phase 3: Add CLI Installation (User Story 1)
+
+- Add step to install CLI via official installer
+- Add step to verify CLI installation
+- Update PATH configuration
+- Move recipe publishing steps to tests job
```

Comment:
Could you please share more details on this? What do we mean by tests job?

Reply:
There is a workflow job named "tests". The recipe publishing would happen there if the build job is removed.